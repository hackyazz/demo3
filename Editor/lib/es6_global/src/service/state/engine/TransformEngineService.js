

import * as TransformAPI$Wonderjs from "../../../../../../node_modules/wonder.js/lib/es6_global/src/api/TransformAPI.js";
import * as HierachyTransformService$Wonderjs from "../../../../../../node_modules/wonder.js/lib/es6_global/src/service/record/main/transform/HierachyTransformService.js";
import * as RecordTransformMainService$Wonderjs from "../../../../../../node_modules/wonder.js/lib/es6_global/src/service/state/main/transform/RecordTransformMainService.js";

function setLocalPosition(localPosition, transform, engineState) {
  return TransformAPI$Wonderjs.setTransformLocalPosition(transform, localPosition, engineState);
}

function setTransformLocalEulerAngles(localEulerAngles, transform, engineState) {
  return TransformAPI$Wonderjs.setTransformLocalEulerAngles(transform, localEulerAngles, engineState);
}

function getParent(child, state) {
  return HierachyTransformService$Wonderjs.getParent(child, RecordTransformMainService$Wonderjs.getRecord(state));
}

var setParent = TransformAPI$Wonderjs.setTransformParent;

var setParentKeepOrder = TransformAPI$Wonderjs.setTransformParentKeepOrder;

var getChildren = TransformAPI$Wonderjs.unsafeGetTransformChildren;

function setLocalEulerAngles(value, component, engineState) {
  return TransformAPI$Wonderjs.setTransformLocalEulerAngles(component, value, engineState);
}

function setLocalScale(value, component, engineState) {
  return TransformAPI$Wonderjs.setTransformLocalScale(component, value, engineState);
}

var getLocalToWorldMatrixTypeArray = TransformAPI$Wonderjs.getTransformLocalToWorldMatrixTypeArray;

var getPosition = TransformAPI$Wonderjs.getTransformPosition;

var setPosition = TransformAPI$Wonderjs.setTransformPosition;

var getLocalPosition = TransformAPI$Wonderjs.getTransformLocalPosition;

var changeChildOrder = TransformAPI$Wonderjs.changeChildOrder;

var getGameObjectByTransform = TransformAPI$Wonderjs.unsafeGetTransformGameObject;

var getLocalRotation = TransformAPI$Wonderjs.getTransformLocalRotation;

var setLocalRotation = TransformAPI$Wonderjs.setTransformLocalRotation;

var getRotation = TransformAPI$Wonderjs.getTransformRotation;

var setRotation = TransformAPI$Wonderjs.setTransformRotation;

var getLocalEulerAngles = TransformAPI$Wonderjs.getTransformLocalEulerAngles;

var getEulerAngles = TransformAPI$Wonderjs.getTransformEulerAngles;

var setEulerAngles = TransformAPI$Wonderjs.setTransformEulerAngles;

var getLocalScale = TransformAPI$Wonderjs.getTransformLocalScale;

var getScale = TransformAPI$Wonderjs.getTransformScale;

var setScale = TransformAPI$Wonderjs.setTransformScale;

var lookAt = TransformAPI$Wonderjs.lookAt;

var rotateLocalOnAxis = TransformAPI$Wonderjs.rotateLocalOnAxis;

var rotateWorldOnAxis = TransformAPI$Wonderjs.rotateWorldOnAxis;

export {
  getPosition ,
  setPosition ,
  getLocalPosition ,
  setLocalPosition ,
  setTransformLocalEulerAngles ,
  getParent ,
  setParent ,
  setParentKeepOrder ,
  getChildren ,
  changeChildOrder ,
  getGameObjectByTransform ,
  getLocalRotation ,
  setLocalRotation ,
  getRotation ,
  setRotation ,
  getLocalEulerAngles ,
  setLocalEulerAngles ,
  getEulerAngles ,
  setEulerAngles ,
  getLocalScale ,
  setLocalScale ,
  getScale ,
  setScale ,
  getLocalToWorldMatrixTypeArray ,
  lookAt ,
  rotateLocalOnAxis ,
  rotateWorldOnAxis ,
  
}
/* TransformAPI-Wonderjs Not a pure module */
