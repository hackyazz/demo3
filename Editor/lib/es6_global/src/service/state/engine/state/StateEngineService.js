

import * as StateAPI$Wonderjs from "../../../../../../../node_modules/wonder.js/lib/es6_global/src/api/StateAPI.js";

var deepCopyForRestore = StateAPI$Wonderjs.deepCopyForRestore;

var createState = StateAPI$Wonderjs.createState;

var createStateData = StateAPI$Wonderjs.createStateData;

var getStateFromData = StateAPI$Wonderjs.getStateFromData;

var unsafeGetState = StateAPI$Wonderjs.unsafeGetState;

var setStateToData = StateAPI$Wonderjs.setStateToData;

var setState = StateAPI$Wonderjs.setState;

var setIsDebug = StateAPI$Wonderjs.setIsDebug;

export {
  deepCopyForRestore ,
  createState ,
  createStateData ,
  getStateFromData ,
  unsafeGetState ,
  setStateToData ,
  setState ,
  setIsDebug ,
  
}
/* StateAPI-Wonderjs Not a pure module */
