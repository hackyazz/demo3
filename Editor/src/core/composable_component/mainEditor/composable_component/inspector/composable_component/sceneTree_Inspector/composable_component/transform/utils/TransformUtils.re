let truncateTransformValue = ((x, y, z)) => {
  let truncateLen = 5;
  (
    FloatService.truncateFloatValue(x, truncateLen),
    FloatService.truncateFloatValue(y, truncateLen),
    FloatService.truncateFloatValue(z, truncateLen),
  );
};

let getSceneTreeNodeLocalPosition = (transformComponent, engineState) =>
  TransformEngineService.getLocalPosition(transformComponent, engineState);

let getTransformPositionData = (transformComponent, engineState) =>
  getSceneTreeNodeLocalPosition(transformComponent, engineState)
  |> truncateTransformValue;

let getTransformScaleData = (transformComponent, engineState) =>
  TransformEngineService.getLocalScale(transformComponent, engineState)
  |> truncateTransformValue;

let getTransformRotationData = (transformComponent, engineState) => {
  let (data, editorState) =
    TransformEditorService.getLocalEulerAngleAndInit(
      transformComponent,
      (StateEditorService.getState(), engineState),
    );

  editorState |> StateEditorService.setState |> ignore;

  data |> truncateTransformValue;
};

let refreshTransformWithDispatchFunc =
    (dispatchFunc, (editorState, engineState)) => {
  let editorState =
    TransformEditorService.removeLocalEulerAngleData(
      GameObjectComponentEngineService.unsafeGetTransformComponent(
        SceneTreeEditorService.unsafeGetCurrentSceneTreeNode(editorState),
        engineState,
      ),
      editorState,
    );

  editorState |> StateEditorService.setState |> ignore;
  engineState |> StateEngineService.setState |> ignore;

  dispatchFunc(AppStore.UpdateAction(Update([|UpdateStore.Inspector|])))
  |> ignore;

  (StateEditorService.getState(), StateEngineService.unsafeGetState());
};

let refreshTransform = ((editorState, engineState)) =>
  refreshTransformWithDispatchFunc(
    UIStateService.getDispatch(),
    (editorState, engineState),
  );