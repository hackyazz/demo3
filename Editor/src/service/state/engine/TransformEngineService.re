/* open Wonderjs; */

/* open TransformType; */

let getPosition = Wonderjs.TransformAPI.getTransformPosition;

let setPosition = Wonderjs.TransformAPI.setTransformPosition;

let getLocalPosition = Wonderjs.TransformAPI.getTransformLocalPosition;

let setLocalPosition = (localPosition, transform, engineState) =>
  Wonderjs.TransformAPI.setTransformLocalPosition(
    transform,
    localPosition,
    engineState,
  );

let setTransformLocalEulerAngles = (localEulerAngles, transform, engineState) =>
  Wonderjs.TransformAPI.setTransformLocalEulerAngles(
    transform,
    localEulerAngles,
    engineState,
  );

let getParent = (child: Wonderjs.TransformType.transform, state) =>
  Wonderjs.HierachyTransformService.getParent(
    child,
    Wonderjs.RecordTransformMainService.getRecord(state),
  );

let setParent =
    (
      parent: Wonderjs.TransformType.transform,
      child: Wonderjs.TransformType.transform,
      state,
    ) =>
  Wonderjs.TransformAPI.setTransformParent(
    Js.Nullable.return(parent),
    child,
    state,
  );

let setParentKeepOrder =
    (
      parent: Wonderjs.TransformType.transform,
      child: Wonderjs.TransformType.transform,
      state,
    ) =>
  Wonderjs.TransformAPI.setTransformParentKeepOrder(
    Js.Nullable.return(parent),
    child,
    state,
  );

let getChildren = (transform: Wonderjs.TransformType.transform, state) =>
  Wonderjs.TransformAPI.unsafeGetTransformChildren(transform, state);

let changeChildOrder = Wonderjs.TransformAPI.changeChildOrder;

let getGameObjectByTransform = Wonderjs.TransformAPI.unsafeGetTransformGameObject;

let getLocalRotation = Wonderjs.TransformAPI.getTransformLocalRotation;

let setLocalRotation = Wonderjs.TransformAPI.setTransformLocalRotation;

let getRotation = Wonderjs.TransformAPI.getTransformRotation;

let setRotation = Wonderjs.TransformAPI.setTransformRotation;

let getLocalEulerAngles = Wonderjs.TransformAPI.getTransformLocalEulerAngles;

let setLocalEulerAngles = (value, component, engineState) =>
  Wonderjs.TransformAPI.setTransformLocalEulerAngles(
    component,
    value,
    engineState,
  );

let getEulerAngles = Wonderjs.TransformAPI.getTransformEulerAngles;

let setEulerAngles = Wonderjs.TransformAPI.setTransformEulerAngles;

let getLocalScale = Wonderjs.TransformAPI.getTransformLocalScale;

let setLocalScale = (value, component, engineState) =>
  Wonderjs.TransformAPI.setTransformLocalScale(component, value, engineState);

let getScale = Wonderjs.TransformAPI.getTransformScale;

let setScale = Wonderjs.TransformAPI.setTransformScale;

let getLocalToWorldMatrixTypeArray = (transform, engineState) =>
  Wonderjs.TransformAPI.getTransformLocalToWorldMatrixTypeArray(
    transform,
    engineState,
  );

let lookAt = Wonderjs.TransformAPI.lookAt;

let rotateLocalOnAxis = Wonderjs.TransformAPI.rotateLocalOnAxis;

let rotateWorldOnAxis = Wonderjs.TransformAPI.rotateWorldOnAxis;