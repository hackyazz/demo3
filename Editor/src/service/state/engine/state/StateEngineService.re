let deepCopyForRestore = Wonderjs.StateAPI.deepCopyForRestore;

let createState = Wonderjs.StateAPI.createState;

let createStateData = Wonderjs.StateAPI.createStateData;

let getStateFromData = Wonderjs.StateAPI.getStateFromData;

let unsafeGetState = Wonderjs.StateAPI.unsafeGetState;

let setStateToData = Wonderjs.StateAPI.setStateToData;

let setState = Wonderjs.StateAPI.setState;

let setIsDebug = Wonderjs.StateAPI.setIsDebug;