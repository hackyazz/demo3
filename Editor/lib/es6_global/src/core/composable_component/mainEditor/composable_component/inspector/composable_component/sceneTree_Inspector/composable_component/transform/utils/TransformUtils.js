

import * as Curry from "../../../../../../../../../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as AppStore$WonderEditor from "../../../../../../../../../ui/store/AppStore.js";
import * as FloatService$WonderEditor from "../../../../../../../../../../service/atom/FloatService.js";
import * as UIStateService$WonderEditor from "../../../../../../../../../../service/state/ui/UIStateService.js";
import * as StateEditorService$WonderEditor from "../../../../../../../../../../service/state/editor/StateEditorService.js";
import * as StateEngineService$WonderEditor from "../../../../../../../../../../service/state/engine/state/StateEngineService.js";
import * as SceneTreeEditorService$WonderEditor from "../../../../../../../../../../service/state/editor/sceneTree/SceneTreeEditorService.js";
import * as TransformEditorService$WonderEditor from "../../../../../../../../../../service/state/editor/transform/TransformEditorService.js";
import * as TransformEngineService$WonderEditor from "../../../../../../../../../../service/state/engine/TransformEngineService.js";
import * as GameObjectComponentEngineService$WonderEditor from "../../../../../../../../../../service/state/engine/gameObject/GameObjectComponentEngineService.js";

function truncateTransformValue(param) {
  return /* tuple */[
          FloatService$WonderEditor.truncateFloatValue(param[0], 5),
          FloatService$WonderEditor.truncateFloatValue(param[1], 5),
          FloatService$WonderEditor.truncateFloatValue(param[2], 5)
        ];
}

var getSceneTreeNodeLocalPosition = TransformEngineService$WonderEditor.getLocalPosition;

function getTransformPositionData(transformComponent, engineState) {
  return truncateTransformValue(TransformEngineService$WonderEditor.getLocalPosition(transformComponent, engineState));
}

function getTransformScaleData(transformComponent, engineState) {
  return truncateTransformValue(TransformEngineService$WonderEditor.getLocalScale(transformComponent, engineState));
}

function getTransformRotationData(transformComponent, engineState) {
  var match = TransformEditorService$WonderEditor.getLocalEulerAngleAndInit(transformComponent, /* tuple */[
        StateEditorService$WonderEditor.getState(/* () */0),
        engineState
      ]);
  StateEditorService$WonderEditor.setState(match[1]);
  return truncateTransformValue(match[0]);
}

function refreshTransformWithDispatchFunc(dispatchFunc, param) {
  var engineState = param[1];
  var editorState = param[0];
  var editorState$1 = TransformEditorService$WonderEditor.removeLocalEulerAngleData(GameObjectComponentEngineService$WonderEditor.unsafeGetTransformComponent(SceneTreeEditorService$WonderEditor.unsafeGetCurrentSceneTreeNode(editorState), engineState), editorState);
  StateEditorService$WonderEditor.setState(editorState$1);
  StateEngineService$WonderEditor.setState(engineState);
  Curry._1(dispatchFunc, [
        AppStore$WonderEditor.UpdateAction,
        /* Update */[/* array */[/* Inspector */2]]
      ]);
  return /* tuple */[
          StateEditorService$WonderEditor.getState(/* () */0),
          StateEngineService$WonderEditor.unsafeGetState(/* () */0)
        ];
}

function refreshTransform(param) {
  return refreshTransformWithDispatchFunc(UIStateService$WonderEditor.getDispatch(/* () */0), /* tuple */[
              param[0],
              param[1]
            ]);
}

export {
  truncateTransformValue ,
  getSceneTreeNodeLocalPosition ,
  getTransformPositionData ,
  getTransformScaleData ,
  getTransformRotationData ,
  refreshTransformWithDispatchFunc ,
  refreshTransform ,
  
}
/* AppStore-WonderEditor Not a pure module */
