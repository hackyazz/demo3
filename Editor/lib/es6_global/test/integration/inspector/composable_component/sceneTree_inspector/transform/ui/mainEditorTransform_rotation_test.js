

import * as Curry from "../../../../../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as Sinon from "../../../../../../../../../node_modules/wonder-bs-sinon/lib/es6_global/src/sinon.js";
import * as Sinon$1 from "sinon";
import * as Wonder_jest from "../../../../../../../../../node_modules/wonder-bs-jest/lib/es6_global/src/wonder_jest.js";
import * as TestTool$WonderEditor from "../../../../../../tool/TestTool.js";
import * as ReactTestTool$WonderEditor from "../../../../../../tool/ReactTestTool.js";
import * as TypeArrayTool$WonderEditor from "../../../../../redo_undo/tool/TypeArrayTool.js";
import * as GameObjectTool$WonderEditor from "../../../../../../tool/GameObjectTool.js";
import * as TransformUtils$WonderEditor from "../../../../../../../src/core/composable_component/mainEditor/composable_component/inspector/composable_component/sceneTree_Inspector/composable_component/transform/utils/TransformUtils.js";
import * as BuildComponentTool$WonderEditor from "../../../../../../tool/BuildComponentTool.js";
import * as StateEngineService$WonderEditor from "../../../../../../../src/service/state/engine/StateEngineService.js";
import * as MainEditorSceneTool$WonderEditor from "../../../../../../tool/MainEditorSceneTool.js";
import * as TransformEngineService$WonderEditor from "../../../../../../../src/service/state/engine/TransformEngineService.js";
import * as MainEditorTransformTool$WonderEditor from "../tool/MainEditorTransformTool.js";

describe("MainEditorTransform rotation", (function () {
        var sandbox = Sinon.getSandboxDefaultVal(/* () */0);
        beforeEach((function () {
                sandbox[0] = Sinon$1.sandbox.create();
                return /* () */0;
              }));
        afterEach((function () {
                return Curry._1(Sinon.restoreSandbox, sandbox[0]);
              }));
        describe("test change rotation value", (function () {
                beforeEach((function () {
                        MainEditorSceneTool$WonderEditor.initState(sandbox, undefined, undefined, /* () */0);
                        return MainEditorSceneTool$WonderEditor.createDefaultScene(sandbox, MainEditorSceneTool$WonderEditor.setFirstCubeToBeCurrentSceneTreeNode);
                      }));
                describe("test change localEulerAngle x", (function () {
                        Wonder_jest.test("test snapshot", (function () {
                                var currentGameObjectTransform = GameObjectTool$WonderEditor.getCurrentSceneTreeNodeTransform(/* () */0);
                                MainEditorTransformTool$WonderEditor.changeRotationX(currentGameObjectTransform, -10.1213);
                                return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildMainEditorTransformComponent(TestTool$WonderEditor.buildEmptyAppState(/* () */0), currentGameObjectTransform));
                              }));
                        return Wonder_jest.test("set to engine state", (function () {
                                      var currentGameObjectTransform = GameObjectTool$WonderEditor.getCurrentSceneTreeNodeTransform(/* () */0);
                                      MainEditorTransformTool$WonderEditor.changeRotationX(currentGameObjectTransform, -10.1213);
                                      var engineState = StateEngineService$WonderEditor.unsafeGetState(/* () */0);
                                      var match = TransformEngineService$WonderEditor.getLocalEulerAngles(currentGameObjectTransform, engineState);
                                      return Curry._2(Wonder_jest.Expect[/* Operators */23][/* = */5], Wonder_jest.Expect[/* expect */0](/* tuple */[
                                                      TypeArrayTool$WonderEditor.truncateFloatValue(5, match[0]),
                                                      match[1],
                                                      match[2]
                                                    ]), /* tuple */[
                                                  -10.1213,
                                                  0,
                                                  0
                                                ]);
                                    }));
                      }));
                describe("deal with the specific case", (function () {
                        beforeEach((function () {
                                var currentGameObjectTransform = GameObjectTool$WonderEditor.getCurrentSceneTreeNodeTransform(/* () */0);
                                TransformUtils$WonderEditor.getTransformRotationData(currentGameObjectTransform);
                                return /* () */0;
                              }));
                        Wonder_jest.test("\n      1.getTransformRotationData;\n      2.set rotation to (x:-45.0, y:180.0, z:0.0);\n\n      inspector->transform->rotation should show (-45.0, 180.0, 0.0)\n      ", (function () {
                                var currentGameObjectTransform = GameObjectTool$WonderEditor.getCurrentSceneTreeNodeTransform(/* () */0);
                                MainEditorTransformTool$WonderEditor.changeRotationX(currentGameObjectTransform, -45);
                                MainEditorTransformTool$WonderEditor.changeRotationY(currentGameObjectTransform, 180);
                                MainEditorTransformTool$WonderEditor.changeRotationZ(currentGameObjectTransform, -0);
                                return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildMainEditorTransformComponent(TestTool$WonderEditor.buildEmptyAppState(/* () */0), currentGameObjectTransform));
                              }));
                        return Wonder_jest.test("\n      1.getTransformRotationData;\n      2.set rotation to (x:-45.0, y:180.0);\n\n      inspector->transform->rotation should show (-45.0, 180.0, 0.0)\n      ", (function () {
                                      var currentGameObjectTransform = GameObjectTool$WonderEditor.getCurrentSceneTreeNodeTransform(/* () */0);
                                      MainEditorTransformTool$WonderEditor.changeRotationX(currentGameObjectTransform, -45);
                                      MainEditorTransformTool$WonderEditor.changeRotationY(currentGameObjectTransform, 180);
                                      return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildMainEditorTransformComponent(TestTool$WonderEditor.buildEmptyAppState(/* () */0), currentGameObjectTransform));
                                    }));
                      }));
                return /* () */0;
              }));
        return /* () */0;
      }));

export {
  
}
/*  Not a pure module */
