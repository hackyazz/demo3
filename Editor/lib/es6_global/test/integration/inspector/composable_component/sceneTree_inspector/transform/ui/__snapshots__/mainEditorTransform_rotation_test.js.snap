// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainEditorTransform rotation test change rotation value deal with the specific case 
      1.getTransformRotationData;
      2.set rotation to (x:-45.0, y:180.0);

      inspector->transform->rotation should show (-45.0, 180.0, 0.0)
       1`] = `
<article
  className="wonder-inspector-transform"
>
  <article
    className="inspector-item wonder-three-float-input"
  >
    <div
      className="item-header"
    >
      Position
    </div>
    <div
      className="item-content"
    >
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          X
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          Y
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          Z
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
    </div>
  </article>
  <article
    className="inspector-item wonder-three-float-input"
  >
    <div
      className="item-header"
    >
      Rotation
    </div>
    <div
      className="item-content"
    >
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          X
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="135"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          Y
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          Z
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="180"
          />
        </div>
      </article>
    </div>
  </article>
  <article
    className="inspector-item wonder-three-float-input"
  >
    <div
      className="item-header"
    >
      Scale
    </div>
    <div
      className="item-content"
    >
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          X
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="1"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          Y
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="1"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          Z
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="1"
          />
        </div>
      </article>
    </div>
  </article>
</article>
`;

exports[`MainEditorTransform rotation test change rotation value deal with the specific case 
      1.getTransformRotationData;
      2.set rotation to (x:-45.0, y:180.0, z:0.0);

      inspector->transform->rotation should show (-45.0, 180.0, 0.0)
       1`] = `
<article
  className="wonder-inspector-transform"
>
  <article
    className="inspector-item wonder-three-float-input"
  >
    <div
      className="item-header"
    >
      Position
    </div>
    <div
      className="item-content"
    >
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          X
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          Y
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          Z
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
    </div>
  </article>
  <article
    className="inspector-item wonder-three-float-input"
  >
    <div
      className="item-header"
    >
      Rotation
    </div>
    <div
      className="item-content"
    >
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          X
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="-45"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          Y
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="180"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          Z
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
    </div>
  </article>
  <article
    className="inspector-item wonder-three-float-input"
  >
    <div
      className="item-header"
    >
      Scale
    </div>
    <div
      className="item-content"
    >
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          X
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="1"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          Y
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="1"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          Z
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="1"
          />
        </div>
      </article>
    </div>
  </article>
</article>
`;

exports[`MainEditorTransform rotation test change rotation value test change localEulerAngle x test snapshot 1`] = `
<article
  className="wonder-inspector-transform"
>
  <article
    className="inspector-item wonder-three-float-input"
  >
    <div
      className="item-header"
    >
      Position
    </div>
    <div
      className="item-content"
    >
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          X
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          Y
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          Z
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
    </div>
  </article>
  <article
    className="inspector-item wonder-three-float-input"
  >
    <div
      className="item-header"
    >
      Rotation
    </div>
    <div
      className="item-content"
    >
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          X
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="-10.1213"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          Y
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          Z
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </div>
      </article>
    </div>
  </article>
  <article
    className="inspector-item wonder-three-float-input"
  >
    <div
      className="item-header"
    >
      Scale
    </div>
    <div
      className="item-content"
    >
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          X
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="1"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          Y
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="1"
          />
        </div>
      </article>
      <article
        className="inspector-item wonder-float-input"
      >
        <div
          className="item-header component-label"
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          onMouseUp={[Function]}
        >
          Z
        </div>
        <div
          className="item-content"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="1"
          />
        </div>
      </article>
    </div>
  </article>
</article>
`;
